### PR TITLE
Add heptio-authenticator-aws plugin

### PR DESCRIPTION
**Pull request by neerfri**
Adding a plugin for [heptio-authenticator-aws](https://github.com/heptio/authenticator)

### CODE REVIEW

**Comment by neerfri** [comment]
@Stratus3D I guess there is not much I can do for the failing tests due to other plugins, right?

File: plugins/heptio-authenticator-aws [hunk 1]
```diff
@@ -0,0 +1 @@
+repository = https://github.com/neerfri/asdf-heptio-authenticator-aws
```
Comment (Stratus3D) [inline plugins/heptio-authenticator-aws]: This should be the URL to the Git repository. e.g. https://github.com/neerfri/asdf-heptio-authenticator-aws.git

**Comment by Stratus3D** [comment]
@neerfri correct. The build is failing but it's unrelated to your PR.

**Comment by neerfri** [comment]
@Stratus3D Fixed

**Comment by deiwin** [comment]
The project's been renamed to `aws-iam-authenticator` and has moved to https://github.com/kubernetes-sigs/aws-iam-authenticator. The plugin still works as is, but would be prudent to update, imo.

**Pull request merged**